tutorials/dataframe/df034_SaveGraph.C

```diff
@@ -0,0 +1,42 @@
+/*
+Use SaveGraph to inspect the sequence of RDataFrame actions.
+
+The tutorial shows how to use SaveGraph action.
+*/
+
+void df034_SaveGraph()
+{
+   ROOT::RDataFrame rd1(1);
+   auto root = rd1.Define("Root_def1", "1").Filter("Root_def1 < 2", "Main_Filter").Define("Root_def2", "1");
+
+   auto branch1 = root.Define("Branch_1_def", "1");
+   auto branch2 = root.Define("Branch_2_def", "1");
+
+   ROOT::RDF::RResultPtr<double> branch1_1 = branch1.Filter("Branch_1_def < 2", "Filter_1")
+                                                .Define("Branch_1_1_def", "1")
+                                                .Filter("1 == Branch_1_1_def % 2", "Filter_1_1")
+                                                .Mean("Branch_1_1_def");
+
+   ROOT::RDF::RResultPtr<unsigned long long> branch1_2 =
+      branch1.Define("Branch_1_2_def", "1").Filter("Branch_1_2_def < 2", "Filter_1_2").Count();
+
+   ROOT::RDF::RResultPtr<double> branch2_1 = branch2.Filter("Branch_2_def < 2", "Filter_2")
+                                                .Define("Branch_2_1_def", "1")
+                                                .Define("Branch_2_2_def", "1")
+                                                .Filter("1 == Branch_2_1_def % 2", "Filter_2_1")
+                                                .Max("Branch_2_1_def");
+
+   ROOT::RDF::RResultPtr<unsigned long long> branch2_2 = branch2.Count();
+
+   /*
+   First, some RDataFrame computation graph is created with Defines, Filters and methods (such as Mean, Count, etc.).
+
+   After that SaveGraph can be called either on the RDataFrame, or alternatively on a specific branch.
+   Moreover, if SaveGraph is called with 1 argument it simply returns a string.
+   And if SaveGraph is called with 2 argument, then it write to a file (passed as the second argument).
+   */
+
+   ROOT::RDF::SaveGraph(rd1, "macro1.dot");
+   gSystem->Exec("dot -Tpng macro1.dot -o macro1.png");
+   /* Independently of SaveGraph, we can convert the produced .dot file to corresponding .png. */
```

Comment:
```suggestion
   ROOT::RDF::SaveGraph(rd1, /*output_file=*/"rdf_savegraph_tutorial.dot");
   // SaveGraph produces content in the standard DOT file format (https://en.wikipedia.org/wiki/DOT_%28graph_description_language%29):
   // it can be converted to e.g. an image file using standard tools such as the `dot` CLI program.
   gSystem->Exec("dot -Tpng rdf_savegraph_tutorial.dot -o rdf_savegraph_tutorial.png");
```